Hardware/Nic/NicGroup.cs

```diff
@@ -20,10 +22,39 @@ public NicGroup(ISettings settings)
 
         private void ScanNics(ISettings settings)
         {
-            NetworkInterfaces = NetworkInterface.GetAllNetworkInterfaces();
-            _hardware = NetworkInterfaces.Where(DesiredNetworkType)
-                .Select((x, i) => new Nic(x, settings, i))
-                .ToList();
+            // If no network is marked up (excluding loopback and tunnel) then don't scan
+            // for interfaces.
+            if (!NetworkInterface.GetIsNetworkAvailable())
+            {
+                return;
+            }
+
+            // When multiple events fire concurrently, we don't want threads interferring
+            // with others as they manipulate non-thread safe state.
+            lock (_scanLock)
+            {
+                NetworkInterfaces = NetworkInterface.GetAllNetworkInterfaces()

```

Comment:
I don't think NetworkInterfaces us actually used anywhere so can probably be removed.

Reply:
The `NetworkInterfaces` property is used in `GetReport` (but only as a null check). Would you like me to remove that null check too?

Reply:
Yes, that can be removed, as _hardware is never null anyway 👍

Reply:
Rest looks good, I'll merge after this change.

Reply:
Excellent, should be resolved